### DISCUSSION TITLE
2021-12-10 Core Tools Update

### DISCUSSION THREAD

**Discussion by austinabell** [near/community#146] [Developer Tools]
### near-sdk-rs
- @austinabell Implemented updated version of `TreeMap` collection (https://github.com/near/near-sdk-rs/pull/665)
- @itegulov Optimized `LookupSet` implementation (https://github.com/near/near-sdk-rs/pull/664)
- @itegulov Implemented safe API for ecrecover (https://github.com/near/near-sdk-rs/pull/658)

### workspaces-rs
- @ChaoticTempest Updated dependencies to use released versions and API cleanup (https://github.com/near/workspaces-rs/pull/25)
- @ChaoticTempest  Updating call API for interacting with contracts (https://github.com/near/workspaces-rs/pull/33)
- @ChaoticTempest Spooning state example of ref-finance

### nearcore
 - @miraclx Updated error type for account id (https://github.com/near/nearcore/pull/5660)
 - @miraclx Implemented workspace crate compliance checks on CI (https://github.com/near/nearcore/pull/5724)
 - @miraclx Opened tracking issue for ensuring all crates are eventually compliant with the defined rules (https://github.com/near/nearcore/issues/5763)

### near-cli-rs
- @frol Added windows support (https://github.com/near/near-cli-rs/pull/78)

**Comment by frol** [comment]
Well, regarding near-cli-rs, I added CI which in addition to the old set of systems (Linux x86-64 [glibc] and Darwin x86-64) compiles the artifacts for Linux ARM and Linux x86-64 [musl - static binary], Windows x86-64 and i686. The mentioned PR was a bit of a false start where I created and merged PR to a wrong fork, oops :fearful: Anyway, it was almost ready by that time, so I finished the remaining bits in the next PR.

**Reply by austinabell** [reply]
Thank you for the clarification!
